### PR TITLE
fix: auto-generate signing keys when missing

### LINKED ISSUE #506: Fix: Prevent HttpsError when no signing keys exist
### Description
The `getActiveKey()` function in `crypto.ts` throws an `HttpsError` when the `keys` collection is empty. This causes tests to fail on fresh databases and could break first-time deployments.

**Current behavior:**
- Empty database → `getActiveKey()` throws HttpsError

### Acceptance Criteria
- [ ] Tests run successfully without manual key creation
- [ ] Clear error message when keys are missing in production
- [ ] No race conditions if multiple requests happen simultaneously
- [ ] Existing functionality remains unchanged
- [ ] Solution is safe for production use

### PR DESCRIPTION
**Pull request by Hassan372**
## Purpose of the change
Fixes the `HttpsError` that crashes when the signing keys collection is empty. The backend now auto-generates keys when needed, and tests automatically seed keys during setup.

## Related issues
Closes #506

### Backend (crypto.ts)
When `getActiveKey()` finds an empty collection, it now generates a new Ed25519 key pair instead of throwing an error. The key gets saved to Firestore with `active=true` and a timestamp. Added logging for visibility.

### Tests - Centralized key seeding
**`FirebaseEmulator.kt`:**
- Added `ensureSigningKeyExists()` to centralize key seeding logic
- Uses a `keySeeded` flag to avoid redundant database operations
- Automatically creates a test Ed25519 key pair when the collection is empty

**Test base classes:**
- `PassFirestoreTestBase.kt`: Calls `FirebaseEmulator.ensureSigningKeyExists()` in `setUp()`
- `FirestoreTestBase.kt`: Also calls the centralized seeding function
- Both classes now share the same key initialization logic instead of duplicating it

## Why it works
- Backend handles production edge cases (first deployment, accidentally deleted keys)
- Tests use a single centralized utility for key setup, avoiding code duplication
- Existing keys are always used first; new ones only created when the collection is empty
- All tests that inherit from either base class automatically have signing keys available

---
**Note:** An LLM was used for code generation and structuring this PR.

### CODE REVIEW

File: app/src/androidTest/java/ch/onepass/onepass/model/pass/PassFirestoreTestBase.kt [hunk 1]
```diff
@@ -77,6 +96,30 @@ open class PassFirestoreTestBase {
...
+            mapOf(
+                "kid" to "test-key-01",
+                "publicKey" to "sMJlPpZyv1oNbluv+zOHhzFKpeVbAWsqKEMgyySbhDO=",
+                "privateKey" to
+                    "mXR5sz6O8sRFXASEQjeWKo9yySD36yxXOhi3iaeNO7qwwkg+InK/Wg1+X8DsHQtwgZtYQNx0fA==",
+                "active" to true,
+                "createdAt" to FieldValue.serverTimestamp()))
+        .await()
+    Log.d("PassFirestoreTestBase", "Test signing key created successfully")
+  }
```
Comment (Bouazz-a) [inline app/src/androidTest/java/ch/onepass/onepass/model/pass/PassFirestoreTestBase.kt]: This function is already defined in  `FirebaseEmulators` class

**Review by Bouazz-a — CHANGES_REQUESTED** [review]
This is a very solid change that makes the system more robust and test-friendly. The only thing to improve is duplication: the signing-key seeding logic now lives both in the test base and in the emulator helper. Centralizing that logic in one shared helper would keep things dry and easier to maintain, while preserving all the benefits of the current approach.

**Comment by Hassan372** [comment]
@BruceKZ already fixed this issue. In order this PR won't be merged.
